programs/mpl-core/src/processor/create.rs

```diff
@@ -25,9 +25,41 @@ pub(crate) struct CreateV1Args {
     pub(crate) plugins: Option<Vec<PluginAuthorityPair>>,
 }
 
-pub(crate) fn create<'a>(accounts: &'a [AccountInfo<'a>], args: CreateV1Args) -> ProgramResult {
+#[repr(C)]
+#[derive(BorshSerialize, BorshDeserialize, PartialEq, Eq, Debug, Clone)]
+pub(crate) struct CreateV2Args {
+    pub(crate) data_state: DataState,
+    pub(crate) name: String,
+    pub(crate) uri: String,
+    pub(crate) plugins: Option<Vec<PluginAuthorityPair>>,
+    pub(crate) external_plugins: Option<Vec<ExternalPluginInitInfo>>,
+}
+
+impl From<CreateV1Args> for CreateV2Args {
```

Comment:
Love this!

Reply:
thanks! Yeah I thought this way was drastically simpler than the other way I tried, which was adding an enum and made a bunch of handling/changes everywhere.  I think its safe to assume we can make versions migratable for now.